src/inputs/TagsEditor/TagsEditor.js

```diff
@@ -0,0 +1,191 @@
+import EditIcon from '@mui/icons-material/Edit';
+
+const DEFAULT_LABELS = {
+  header: 'Tags',
+  placeholder: 'Enter a new tag',
+};
+
+const useStyles = makeStyles((theme) => ({
+  label: {
+    color: theme.palette.text.secondary,
+  },
+  addTagIcon: {
+    width: '16px',
+    height: '16px',
+  },
+}));
+
+export const TagsEditor = (props) => {
+  const classes = useStyles();
+  const { id, values, readOnly, onChange, labels: tmpLabels, chipProps, textFieldProps, editIconProps } = props;
+  const labels = { ...DEFAULT_LABELS, ...tmpLabels };
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [isHovered, setIsHovered] = useState(false);
+
+  const startEdition = useCallback((event) => {
+    event.stopPropagation();
+    event.preventDefault();
+    setIsEditing(true);
+  }, []);
+
+  const stopEdition = useCallback((event) => {
+    event.stopPropagation();
+    event.preventDefault();
+    setIsEditing(false);
+  }, []);
+
+  const addNewTag = useCallback(
+    (newTag) => {
+      setIsEditing(false);
+      if (newTag.trim().length === 0) return;
+      onChange([...values, newTag]);
+    },
+    [onChange, values]
+  );
+
+  const deleteTag = useCallback(
+    (removedTagIndex) => {
+      const newTags = [...values];
+      newTags.splice(removedTagIndex, 1);
+      onChange(newTags);
+    },
+    [onChange, values]
+  );
+
+  const newTagInput = useMemo(() => {
+    if (readOnly || !isEditing) return null;
+    return (
+      <TextField
+        data-cy="new-tag-textfield"
+        sx={{ maxWidth: '150px' }}
+        variant="outlined"
+        size="small"
+        onKeyDown={(event) => {
+          if (event.key === 'Enter') {
+            addNewTag(event.target.value);
+            event.preventDefault();
+          } else if (event.key === 'Escape') {
+            stopEdition(event);
+          }
+        }}
+        onBlur={(event) => addNewTag(event.target.value)}
+        placeholder={labels.placeholder}
+        inputRef={(input) => input && input.focus()}
+        inputProps={{ id: 'new-tag-input' }}
+        InputProps={{ sx: { maxHeight: '25px' } }}
+        {...textFieldProps}
+      />
+    );
+  }, [isEditing, addNewTag, readOnly, stopEdition, labels.placeholder, textFieldProps]);
+
+  const addTagIcon = useMemo(() => {
+    if (readOnly || isEditing || !isHovered) return null;
+    return (
+      <EditIcon data-cy="add-tag" className={classes.addTagIcon} onClick={startEdition} {...editIconProps}></EditIcon>
+    );
+  }, [readOnly, isEditing, isHovered, classes.addTagIcon, startEdition, editIconProps]);
+
+  return (
+    <Grid
+      id={id}
+      key={id}
+      container
+      spacing={1}
+      sx={{ flexFlow: 'column nowrap', alignItems: 'stretch' }}
+      onMouseOver={() => setIsHovered(true)}
+      onMouseLeave={() => setIsHovered(false)}
+    >
+      <Grid item>
+        <Typography variant="body1" className={classes.label} sx={{ pr: '8px' }}>
```

Comment:
More of a question : can we use `spacing` values of mui theme in _ui_ package ? Because it works also for padding and margins : `{{ px: 1 }}` :slightly_smiling_face:

Reply:
I think we can use the theme `spacing` values, yes (I guess the mui theme provider can provide the required information to all components, even those coming from our _ui_ package).
By the way, the part `sx={{ pr: '8px' }}` of the line you mentioned is not necessary in this component, I'll remove it. Thank you for the question! :smile: